Shaders/PBR/private/PBR_PrecomputeCommon.fxh

```diff
@@ -22,7 +22,7 @@ float3 ImportanceSampleGGX(float2 Xi, float PerceptualRoughness, float3 N)
     float a2             = AlphaRoughness * AlphaRoughness;
     
     float Phi      = 2.0 * PI * Xi.x;
-    float CosTheta = sqrt( (1.0 - Xi.y) / ( 1.0 + (a2 - 1.0) * Xi.y ) );
+    float CosTheta = max(0.0, sqrt((1.0 - Xi.y) / (1.0 + (a2 - 1.0) * Xi.y)));
```

Comment:
This does not make sense - how square root can be negative? Why compare it to 0?

Reply:
Run GLTFViewer/PostProcessing on Vulkan. It works. Inside square root, there's likely a division by zero happening when a2 == 0 and Xi.y == 1.0, even though the random number generator should give values in the range [0, 1).

Reply:
![image](https://github.com/user-attachments/assets/786d3024-c3d7-438a-b588-3d6802b8b202)

Reply:
The right fix is then to not allow denominator be zero or the argument of sqrt become negative.
Your current fix relies on undefined behavior that `max(0.0, NAN) == 0`

Reply:
After the fix
![image](https://github.com/user-attachments/assets/239962b9-7ab0-43de-b841-e6b61541a7d0)

Reply:
> BRDF_LUT on Vulkan before the fix ![image](https://private-user-images.githubusercontent.com/25492259/357636028-786d3024-c3d7-438a-b588-3d6802b8b202.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM1OTcwNTYsIm5iZiI6MTcyMzU5Njc1NiwicGF0aCI6Ii8yNTQ5MjI1OS8zNTc2MzYwMjgtNzg2ZDMwMjQtYzNkNy00MzhhLWI1ODgtM2Q2ODAyYjhiMjAyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE0VDAwNTIzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZkNTJlY2U3MzJkNDNjM2I1NTk1OGFlMWRkZmVjNThmN2ViZGJjMDUyNzNjM2I5NzcwZmRhYjkwZGFkYjdlNjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vH23hl3qFwD0juzmQ1JSY6wrSUrQplumUZ40Ba-_Xe8)

It is too much error.
It is possible that maybe one pixel-line is computed incorrectly.
There is the whole strip of pixels that is computed incorrectly. There is definitely something wrong that is going on.

Reply:
> The right fix is then to not allow denominator be zero or the argument of sqrt become negative. Your current fix relies on undefined behavior that `max(0.0, NAN) == 0`

Then a bias will be introduced. max(0, NaN) is acceptable for such cases. Overall, as I mentioned before, the error lies in the quasi-random number generator, since Xi.y should be strictly less than 1

Reply:
> > BRDF_LUT on Vulkan before the fix ![image](https://private-user-images.githubusercontent.com/25492259/357636028-786d3024-c3d7-438a-b588-3d6802b8b202.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM1OTcwNTYsIm5iZiI6MTcyMzU5Njc1NiwicGF0aCI6Ii8yNTQ5MjI1OS8zNTc2MzYwMjgtNzg2ZDMwMjQtYzNkNy00MzhhLWI1ODgtM2Q2ODAyYjhiMjAyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE0VDAwNTIzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZkNTJlY2U3MzJkNDNjM2I1NTk1OGFlMWRkZmVjNThmN2ViZGJjMDUyNzNjM2I5NzcwZmRhYjkwZGFkYjdlNjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vH23hl3qFwD0juzmQ1JSY6wrSUrQplumUZ40Ba-_Xe8)
> 
> It is too much error. It is possible that maybe one pixel-line is computed incorrectly. There is the whole strip of pixels that is computed incorrectly. There is definitely something wrong that is going on.

Such behavior occurs only on the NVIDIA driver, we can't know what's really happening there